### PR TITLE
Backport ES changes

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commits

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/BlobHeadRequestHandlerTests.java [hunk 1]
```diff
@@ -45,15 +33,62 @@
...
 
+    private ThreadPool threadPool;
+    private CapturingTransport transport;
+    private ClusterService clusterService;
+    private TransportService transportService;
+
+    @Before
+    public void setUpResources() throws Exception {
+        threadPool = new TestThreadPool(getClass().getName());
+        transport = new CapturingTransport() {
```
Comment (seut) [inline server/src/test/java/io/crate/BlobHeadRequestHandlerTests.java:67]: Is this used? I did not find any usage of it

**Review by seut — APPROVED** [review]
Lgtm otherwise.

File: server/src/test/java/io/crate/BlobHeadRequestHandlerTests.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/BlobHeadRequestHandlerTests.java:67]: Couple lines below:

`transportService = transport.createTransportService(`

but could turn it into a local variable

Comment (seut) [reply server/src/test/java/io/crate/BlobHeadRequestHandlerTests.java:67]: thx. read it multiple times but couldn't find it 🤦 
local var sounds good.

**Pull request merged**